pallets/loans-ref/src/lib.rs

```diff
@@ -0,0 +1,792 @@
+		type LoanId: Parameter
+			+ Member
+			+ MaybeSerializeDeserialize
+			+ Default
+			+ TypeInfo
+			+ MaxEncodedLen
+			+ Copy
+			+ EnsureAdd
+			+ One;
+
+		/// Defines the rate type used for math computations
+		type Rate: Parameter
+			+ Member
+			+ MaybeSerializeDeserialize
+			+ FixedPointNumber
+			+ TypeInfo
+			+ MaxEncodedLen;
+
+		/// Defines the balance type used for math computations
+		type Balance: tokens::Balance + FixedPointOperand;
+
+		/// Fetching method for the time of the current block
+		type Time: UnixTime;
+
+		/// Used to mint, transfer, and inspect assets.
+		type NonFungible: Transfer<Self::AccountId>
+			+ Inspect<Self::AccountId, CollectionId = Self::CollectionId, ItemId = Self::ItemId>;
+
+		/// Access to the pool
+		type Pool: PoolReserve<Self::AccountId, Self::CurrencyId, Balance = Self::Balance>;
+
+		/// Used to verify permissions of users
+		type Permissions: Permissions<
+			Self::AccountId,
+			Scope = PermissionScope<PoolIdOf<Self>, Self::CurrencyId>,
+			Role = Role,
+			Error = DispatchError,
+		>;
+
+		/// Used to calculate interest accrual for debt.
+		type InterestAccrual: InterestAccrual<
+			Self::Rate,
+			Self::Balance,
+			Adjustment<Self::Balance>,
+			NormalizedDebt = Self::Balance,
+		>;
+
+		/// Max number of active loans per pool.
+		#[pallet::constant]
+		type MaxActiveLoansPerPool: Get<u32>;
+
+		/// Max number of write-off groups per pool.
+		#[pallet::constant]
+		type MaxWriteOffGroups: Get<u32>;
+	}
+
+	/// Contains the last loan id generated
+	#[pallet::storage]
+	pub(crate) type LastLoanId<T: Config> =
+		StorageMap<_, Blake2_128Concat, PoolIdOf<T>, T::LoanId, ValueQuery>;
+
+	/// Storage for loans that has been created but are not still active.
+	#[pallet::storage]
+	pub(crate) type CreatedLoans<T: Config> = StorageDoubleMap<
+		_,
+		Blake2_128Concat,
+		PoolIdOf<T>,
+		Blake2_128Concat,
+		T::LoanId,
+		CreatedLoan<T>,
+		OptionQuery,
+	>;
+
+	/// Storage for active loans.
+	/// The indexation of this storage differs from `CreatedLoans` or `ClosedLoans`
+	/// because here we try to minimize the iteration speed over all active loans in a pool.
+	/// `Moment` value along with the `ActiveLoan` correspond to the last moment the active loan was
+	/// used to compute the portfolio valuation in an inexact way.
+	#[pallet::storage]
+	pub(crate) type ActiveLoans<T: Config> = StorageMap<
+		_,
+		Blake2_128Concat,
+		PoolIdOf<T>,
+		BoundedVec<(ActiveLoan<T>, Moment), T::MaxActiveLoansPerPool>,
+		ValueQuery,
+	>;
+
+	/// Storage for closed loans.
+	/// No mutations are expected in this storage.
+	/// Loans are stored here for historical purposes.
+	#[pallet::storage]
+	pub(crate) type ClosedLoans<T: Config> = StorageDoubleMap<
+		_,
+		Blake2_128Concat,
+		PoolIdOf<T>,
+		Blake2_128Concat,
+		T::LoanId,
+		ClosedLoan<T>,
+		OptionQuery,
+	>;
+
+	/// Stores write off policy used in each pool
+	#[pallet::storage]
+	pub(crate) type WriteOffPolicy<T: Config> = StorageMap<
```

Comment:
You're right!

Reply:
Rethinking this, a `Policy` is a Vector of `WriteOffStorage`, so with the above naming style, it should be `WriteOffPolicy`, because you get the vector, therefore, one *policy*.